astropy/visualization/interval.py

```diff
@@ -110,12 +138,7 @@ def get_limits(self, values):
         if self.vmin is not None and self.vmax is not None:
             return self.vmin, self.vmax
 
-        # Make sure values is a Numpy array
-        values = np.asarray(values).ravel()
```

Comment:
I think a simpler `np.asanyarray(...)` here should work, without any further extra code (bit of an aside, but: I don't see why you'd need `.ravel()`).

Reply:
`asanyarray` fails because later functions that operate on the data (like np.percentile) ignore the mask and give warnings:
`UserWarning: Warning: 'partition' will ignore the 'mask' of the MaskedArray.   We really need to extract the non-masked values.

I agree about the `ravel` -- I think we can drop that.

Reply:
@mhvk Should I add support for Astropy `Masked` too?  i.e., `values = values.unmasked[~values.mask]`

Reply:
Ah, I even misread - I thought we were talking about `Masked`, which I think might be able to deal with `percentile`. You could use `astropy.utils.masked.get_data_and_mask` to get the relevant data and mask independent of masked class. (There may be some use in doing it differently by making everything `Masked` and using that for the finite stuff too, but a bit too late here to think that through...)